tests/solidity/test/ComplianceBridge.js

```diff
@@ -1,14 +1,111 @@
-const { expect } = require("chai");
-const { ethers } = require("hardhat")
-const { sendShieldedTransaction, sendShieldedQuery } = require("./testUtils")
+const { expect } = require('chai')
+const { ethers } = require('hardhat')
+const { sendShieldedTransaction, sendShieldedQuery } = require('./testUtils')
+
+const unencryptedProvider = new ethers.providers.JsonRpcProvider('http://localhost:8547') // Unencrypted rpc url
 
 describe('ComplianceBridge', () => {
+    const CONTRACT_ADDRESS = '0x30252afe8c1683fd184c99a3c44aa5d547d59dd4'
     let contract
 
     before(async () => {
         const [signer] = await ethers.getSigners()
         const ComplianceProxyFactory = await ethers.getContractFactory('ComplianceProxy')
-        contract = new ethers.Contract("0x30252afe8c1683fd184c99a3c44aa5d547d59dd4", ComplianceProxyFactory.interface, signer);
+        contract = new ethers.Contract(CONTRACT_ADDRESS, ComplianceProxyFactory.interface, signer)
+    })
+
+    describe('Should not change any state', async () => {
+        it('eth_call', async () => {
+            const [signer] = await ethers.getSigners()
+
+            const wallet = ethers.Wallet.createRandom()
+            if (signer.address === wallet.address) {
```

Comment:
It's very unlikely to generate the same wallet as signer, you can skip this check